### ISSUE TITLE
src_dir does not work if env: name != board name

### ISSUE THREAD

**Issue by tig** [platformio/platformio-core#4012]
What kind of issue is this?

- [ ] **Question**.
      This issue tracker is not the place for questions. If you want to ask how to do something,
      or to understand why something isn't working the way you expect it to, 
      use [Community Forums](https://community.platformio.org) or [Premium Support](https://platformio.org/support)

- [ ] **PlatformIO IDE**.
      All issues related to PlatformIO IDE should be reported to appropriate repository:
      [PlatformIO IDE for Atom](https://github.com/platformio/platformio-atom-ide/issues) or
      [PlatformIO IDE for VSCode](https://github.com/platformio/platformio-vscode-ide/issues)

- [ ] **Development Platform or Board**.
      All issues (building, uploading, adding new boards, etc.) related to PlatformIO development platforms
      should be reported to appropriate repository related to your hardware
      https://github.com/topics/platformio-platform

- [ ] **Feature Request**.
      Start by telling us what problem you’re trying to solve. Often a solution
      already exists! Don’t send pull requests to implement new features without first getting our
      support. Sometimes we leave features out on purpose to keep the project small.

- [ X] **PlatformIO Core**.
      If you’ve found a bug, please provide an information below.

*You can erase any parts of this template not applicable to your Issue.*

------------------------------------------------------------------

### Configuration

**Operating system**: Windows 10

**PlatformIO Version** (`platformio --version`): version 5.1.1

### Description of problem

If the name of an `[env:<name>]` block does not perfectly match the name of the `board =` within that `env` the `src_dir = ` within the `[platformio]` section of `platformio.ini` is ignored.

I spent HOURs trying to figure out why `src_dir` was not working to build my examples in several of my libraries. I had gotten in the habit of changing the `env:<name>`s because  when I'm working on multiple libraries at the same time, it helps to have the platformio build logs show an env name that is specific to the library.

#### Steps to Reproduce

This works:
```
[platformio]
src_dir = ./examples/light_switch
;src_dir = ./examples/multitasking
;src_dir = ./examples/timed_switchoff

[env:megaatmega2560]
platform = atmelavr
board = megaatmega2560
framework = arduino

lib_deps = file://src
```

This does not work:
```
[platformio]
; Uncomment one of the following to pick wwhich example PlatformIO will build & run
src_dir = ./examples/light_switch
;src_dir = ./examples/multitasking
;src_dir = ./examples/timed_switchoff

[env:arduino-fsm-mega2560]
platform = atmelavr
board = megaatmega2560
framework = arduino

lib_deps = file://src
```

**Source file to reproduce issue:**

Clone this and change the `[env:megaatmega2560]` in the `platformio.ini` to `[env:somethingelse]`. Build and you'll note that the example is not found (`setup()` and `loop()` not found by the linker).

```cpp
git clone https://github.com/tig/arduino-fsm 
```

**Comment by valeros** [comment]
Hi @tig ! It's strange, but I cannot reproduce the issue. Tried both Windows and Ubuntu,  everything works as expected even when the env name is not the same the board name (as is in your last-mentioned configuration that wasn't supposed to work)
![image](https://user-images.githubusercontent.com/886732/134481451-90fe7e5a-755f-4964-8416-dd91ab592a9c.png)

Could you please try to upgrade PlatformIO to the latest available 5.2.0? Just run `pio upgrade` in your terminal.

**Comment by ivankravets** [comment]
Please reopen if you still need help
